salt/utils/vmware.py

```diff
@@ -2026,44 +2244,541 @@ def get_hosts(service_instance, datacenter_name=None, host_names=None,
+                                                                storage_system,
+                                                                hostname)
+        disk_keys = [key for scsi_addr, key
+                     in six.iteritems(lun_key_by_scsi_addr)
+                     if scsi_addr in scsi_addresses]
+        log.trace('disk_keys based on scsi_addresses = {0}'.format(disk_keys))
+
+    scsi_luns = get_all_luns(host_ref, storage_system)
+    scsi_disks = [disk for disk in scsi_luns
+                  if isinstance(disk, vim.HostScsiDisk) and (
+                      get_all_disks or
+                      # Filter by canonical name
+                      (disk_ids and (disk.canonicalName in disk_ids)) or
+                      # Filter by disk keys from scsi addresses
+                      (disk.key in disk_keys))]
+    log.trace('Retrieved disks in host \'{0}\': {1}'
+              ''.format(hostname, [d.canonicalName for d in scsi_disks]))
+    return scsi_disks
+
+
+def get_disk_partition_info(host_ref, disk_id, storage_system=None):
+    '''
+    Returns all partitions on a disk
+
+    host_ref
+        The reference of the ESXi host containing the disk
+
+    disk_id
+        The canonical name of the disk whose partitions are to be removed
+
+    storage_system
+        The ESXi host's storage system. Default is None.
+    '''
+    hostname = get_managed_object_name(host_ref)
+    service_instance = get_service_instance_from_managed_object(host_ref)
+    if not storage_system:
+        storage_system = get_storage_system(service_instance, host_ref,
+                                            hostname)
+
+    props = get_properties_of_managed_object(storage_system,
+                                             ['storageDeviceInfo.scsiLun'])
+    if not props.get('storageDeviceInfo.scsiLun'):
+        raise salt.exceptions.VMwareObjectRetrievalError(
+            'No devices were retrieved in host \'{0}\''.format(hostname))
+    log.trace('[{0}] Retrieved {1} devices: {2}'.format(
+        hostname, len(props['storageDeviceInfo.scsiLun']),
+        ', '.join([l.canonicalName
+                   for l in props['storageDeviceInfo.scsiLun']])))
+    disks = [l for l in props['storageDeviceInfo.scsiLun']
+             if isinstance(l, vim.HostScsiDisk) and
+             l.canonicalName == disk_id]
+    if not disks:
+        raise salt.exceptions.VMwareObjectRetrievalError(
+            'Disk \'{0}\' was not found in host \'{1}\''
+            ''.format(disk_id, hostname))
+    log.trace('[{0}] device_path = {1}'.format(hostname, disks[0].devicePath))
+    partition_info = _get_partition_info(storage_system, disks[0].devicePath)
+    log.trace('[{0}] Retrieved {1} partition(s) on disk \'{2}\''
+              ''.format(hostname, len(partition_info.spec.partition), disk_id))
+    return partition_info
+
+
+def erase_disk_partitions(service_instance, host_ref, disk_id,
```

Comment:
I'm not sure I'm all that thrilled to have this sort of destructive behavior even _available_ in Salt. Is this a "just because we can" function or is there a strong use case for this? At the very least, I would like to make sure that the defaults here do as little damage as possible.

Reply:
It is more because I would like everything to be done in salt, making the bulky vsphere client obsolete. This is not included in the state, so I consider it a manual process. Furthermore, one has to invoke `salt <host_id> vsphere.erase_disk_partitions <disk_id>`. I am not sure how to make it more specific than that - it assumes the user knows exactly the disk_id he/she needs and can't run it on more hosts by mistake because the ids are unique.

Reply:
That is a reasonable point. Thanks, @alexbleotu